pennylane/qnn/keras.py

```diff
@@ -0,0 +1,207 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module is for PennyLane-Keras integration."""
+import functools
+import inspect
+from collections.abc import Iterable
+from typing import Optional
+
+try:
+    import tensorflow as tf
+    from tensorflow.keras.layers import Layer
+    from pennylane.interfaces.tf import to_tf
+
+    CORRECT_TF_VERSION = int(tf.__version__.split(".")[0]) > 1
+except ImportError:
+    from abc import ABC
+
+    Layer = ABC
+    CORRECT_TF_VERSION = False
+
+from pennylane.qnodes import QNode
+
+INPUT_ARG = "inputs"
+
+
+class KerasLayer(Layer):
+    """A Keras layer for integrating PennyLane QNodes with the Keras API.
+
+    This class converts a :func:`~.QNode` to a Keras layer. The QNode must have a signature that
+    satisfies the following conditions:
+
+    - Contain an ``inputs`` named argument for input data. All other arguments are treated as
```

Comment:
All inputs need to go through this argument? Can it be a ragged array, in case one needs to feed X and Y information or so?